inlong-manager/manager-service/src/test/java/org/apache/inlong/manager/service/queue/PulsarUtilsTest.java

```diff
@@ -17,39 +17,671 @@
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.ArgumentMatchers.eq;
+import static org.mockito.Mockito.verify;
+import static org.powermock.api.mockito.PowerMockito.when;
 
 /**
  * Test class for Pulsar utils.
  */
+@ExtendWith(MockitoExtension.class)
+@MockitoSettings(strictness = Strictness.LENIENT)
 public class PulsarUtilsTest {
 
+    private static final Gson GSON = new GsonBuilder().create(); // thread safe
+
+    private static final String DEFAULT_SERVICE_URL = "http://127.0.0.1:8080";
+
+    private static final String DEFAULT_TENANT = "public";
+
+    private static final String DEFAULT_NAMESPACE = "default";
+
+    private static final String DEFAULT_TOPIC = "testtopic";
+
+    private static final String DEFAULT_NON_PRTITIONED_TOPIC = "testtopic_np";
+
+    private static final int DEFAULT_PARTITIONS_NUM = 3;
+
+    private static final String DEFAULT_TOPIC_PATH = DEFAULT_TENANT + "/" + DEFAULT_NAMESPACE + "/" + DEFAULT_TOPIC;
+
+    private static final String DEFAULT_NON_PRTITIONED_TOPIC_PATH =
+            DEFAULT_TENANT + "/" + DEFAULT_NAMESPACE + "/" + DEFAULT_NON_PRTITIONED_TOPIC;
+
+    @Mock
+    private RestTemplate restTemplate;
+
+    @Mock
+    private ResponseEntity<String> exchange;
+
+    @Mock
+    private ResponseEntity<byte[]> byteExchange;
+
+    @Mock
+    private PulsarClusterInfo pulsarClusterInfo;
+
+    private static PulsarNamespacePolicies policies;
+
+    @BeforeAll
+    public static void beforeAll() {
+        policies = new PulsarNamespacePolicies();
+        policies.setMessageTtlInSeconds(10);
+        PulsarPersistencePolicies persistencePolicies = new PulsarPersistencePolicies();
+        persistencePolicies.setBookkeeperEnsemble(1);
+        persistencePolicies.setBookkeeperAckQuorum(2);
+        persistencePolicies.setBookkeeperWriteQuorum(3);
+        persistencePolicies.setManagedLedgerMaxMarkDeleteRate(4.0);
+        policies.setPersistence(persistencePolicies);
+        PulsarRetentionPolicies retentionPolicies = new PulsarRetentionPolicies();
+        retentionPolicies.setRetentionSizeInMB(2048l);
+        retentionPolicies.setRetentionTimeInMinutes(500);
+        policies.setRetentionPolicies(retentionPolicies);
+    }
+
+    @BeforeEach
+    public void before() {
+        when(pulsarClusterInfo.getAdminUrl()).thenReturn(DEFAULT_SERVICE_URL);
+        when(pulsarClusterInfo.getToken()).thenReturn("testtoken");
+
+    }
+    /**
+     * Test cases for {@link PulsarUtils#getPulsarClusters}.
+     *
+     * @throws Exception
+     */
+    @Test
+    public void testGetPulsarClusters() throws Exception {
+        final String url = DEFAULT_SERVICE_URL + PulsarUtils.QUERY_CLUSTERS_PATH;
+        final String result = "[\"standalone\"]";
+        List<String> expected = GSON.fromJson(result, ArrayList.class);
+        when(restTemplate.exchange(eq(url), eq(HttpMethod.GET), any(HttpEntity.class), eq(String.class))).thenReturn(
```

Comment:
It's better to run a pulsar broker service in the test, and then use the pulsar service to verify all these HTTP requests work well.